extensions/src/main/java/org/pentaho/platform/plugin/action/ActionInvokerSystemListener.java

```diff
@@ -0,0 +1,155 @@
+/*!
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2017 Pentaho Corporation..  All rights reserved.
+ */
+package org.pentaho.platform.plugin.action;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.pentaho.platform.api.action.IActionInvokeStatus;
+import org.pentaho.platform.api.action.IActionInvoker;
+import org.pentaho.platform.api.engine.IPentahoSession;
+import org.pentaho.platform.api.engine.IPentahoSystemListener;
+import org.pentaho.platform.engine.core.system.PentahoSystem;
+import org.slf4j.MDC;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.io.IOException;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileFilter;
+
+public class ActionInvokerSystemListener implements IPentahoSystemListener {
+
+  private String environmentVariablesFilepath;
+  private static final Log logger = LogFactory.getLog( IPentahoSystemListener.class );
+  private static final String WORK_ITEM_FILE_EXTENSION = ".json";
+  private static final String DEFAULT_CONTENT_FOLDER = "system/default-content";
+
+  private Map<String, JSONObject> parseJsonFiles() throws IOException {
+    Map<String, JSONObject> dotJsons = new HashMap<>(  );
+    final String solutionPath = PentahoSystem.getApplicationContext().getSolutionPath( DEFAULT_CONTENT_FOLDER );
+    if ( solutionPath == null || solutionPath.isEmpty() ) {
+      logger.info( getClass().getName() + " Empty solution path" );
+    } else {
+      File[] jsonFiles = listFiles( new File( solutionPath ), WORK_ITEM_FILE_EXTENSION );
+      if ( jsonFiles != null ) {
+        for ( File jsonFile : jsonFiles ) {
+          JSONObject jsonObject = getJsonFromFile(  jsonFile  );
+          if ( jsonObject != null ) {
+            dotJsons.put( jsonFile.getAbsolutePath(), jsonObject );
+          }
+        }
+      }
+    }
+    return dotJsons;
+  }
+
+  private JSONObject getJsonFromFile( File file ) throws IOException {
```

Comment:
Don't see the purpose for this method, try this one-liner instead

`JSONObject jsonObj = new JSONObject ( IOUtils.toString( new FileInputStream( file ) ) );`